### ISSUE TITLE
enterprise-login and openssh's lack of kerberos support

### ISSUE THREAD

**Issue by adavidzh** [clearlinux/distribution#2104]
### TL;DR
[The `openssh` package is presently compiled without kerberos support](https://github.com/clearlinux-pkgs/openssh/blob/05440f1924e9cada620444f21c181d790d27666c/configure) such that even after installing [`kerberos` with the `enterprise-login` bundle](https://github.com/clearlinux/clr-bundles/blob/8545cb63c80659f47818035d8ff04215caaaf751/bundles/enterprise-login#L14) it is not possible to use `kerberos` for `ssh` authentication. `git` also shows issues, possibly as a knock-on consequence. This can probably be resolved by configuring `openssh` with `--with-kerberos5`, but it is likely that there will implications in terms of dependency management. I am new here and this is issue is something some may see as a bug and others as an enhancement. See discussion at the end.

### Down the rabbit hole

This issue was first reported [in the forums](https://community.clearlinux.org/t/missing-kerberos-support-in-ssh-and-git/5259/5). Kudos to @lebensterben for getting me on the right track.

I've just started with Clear Linux* on bare metal, and it's generally been a very positive experience. AFAICT I'm running the latest version:
```console
$ sudo swupd update
Update started
Version on server (33620) is not newer than system version (33620)
Update complete - System already up-to-date at version 33620
```

That said, I have hit a problem: it seems that the `ssh` and `git` installed via `swupd` cannot make use of kerberos tickets acquired with the `kinit` also installed via `swupd` (`enterprise-login` bundle).

Main symptoms are that I cannot `ssh` or perform `git` operations in cases where, e.g., my laptop running MacOS works fine.

In both cases, I can get a nice kerberos ticket (slightly obfuscated):
- MacOS
    ```console
    Credentials cache: API:C22BC4A1-3B2B-4302-9757
            Principal: x@Y

      Issued                Expires             Flags    Principal
    Aug 20 14:17:01 2020  Aug 21 15:17:01 2020  FPRIA  krbtgt/Y@Y
    ```
- Clear Linux* 33620 looks equally ok:
    ```console
    Ticket cache: FILE:/tmp/krb5cc_1000
    Default principal: x@Y

    Valid starting       Expires              Service principal
    08/20/2020 14:06:22  08/21/2020 15:06:22  krbtgt/Y@Y
    	renew until 08/27/2020 14:06:22, Flags: FPRIA
    ```

Then what happens, e.g., with `ssh`, is that:
- MacOS (works):
    ```console
    debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic,password,keyboard-interactive
    debug1: Next authentication method: gssapi-with-mic
    debug2: we sent a gssapi-with-mic packet, wait for reply
    debug1: Delegating credentials
    debug1: Delegating credentials
    debug1: Authentication succeeded (gssapi-with-mic).
    ```
- Clear Linux* 33620 (fails):
    ```console
    debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic,password,keyboard-interactive
    debug1: Next authentication method: publickey
    ```

Basically, `ssh` in Clear Linux* 33620 does not offer `gssapi-with-mic` and everything goes south from there.

### To bug or not to bug

I don't know what the philosophy regarding dependencies is in Clear Linux* and I can see two possible arguments being made in considering this issue:
1. Installing `enterprise-login` sets an expectation that `ssh` (and perhaps other packages!) will be able to make use of `kerberos`. I filed this issue as a bug because I had this expectation. Basically, is it a bug that Clear Linux* offers `kerberos` but no `openssh` with `kerberos` support?
1. `openssh` and  `kerberos` are their own separate things and there is no issue. I think this makes software interoperability in Clear Linux* poor by design with the consequences that that can have on usability. Basically, should those who install `enterprise-login` figure out their own `ssh` situation if they want `kerberos` support?

### Thoughts on moving forward

I'd think that those who do not install `enterprise-login` should not be burdened with a "bloated" `openssh` that pulls in `kerberos` as a dependency.
That means that we cannot just add `--with-kerberos5` to the `openssh`'s package `configure` script.

OTOH, I think that it makes sense that those who install `enterprise-login` should seamlessly see the "default" `openssh`  "upgraded" to a version that supports `kerberos` (call it `openssh-with-kerberos`).
Basically, I am pretty sure that everyone installing `enterprise-login` would be ok with their `ssh` (and perhaps other packages) being able to make use of `kerberos`.

Of course, even if `swupd` could in some way upgrade/downgrade `openssh` to/from `openssh-with-kerberos` when installing/removing `enterprise-login`, the fact of creating a new package that differs only by a configuration option is probably a big overhead for maintenance, building, and distribution.
Not to mention the case of multiple dependencies; for N dependencies there could be up to 2<sup>N</sup> different binaries to cover all the combinations.

Hopefully this sort of dependency issue has come up before and there is a mechanism to manage it.

Thanks!

**Comment by plispe** [comment]
Hi, any update to this?

**Comment by bryteise** [comment]
Enabling kerberos with openssh hasn't been something we are trying at this point and there isn't a good way to have an alternate ssh for this option either. I'll take a look what gets pulled in and how big that all is when adding the option though.

**Comment by bryteise** [comment]
Looks like 19M increase. I enabled the option.

**Comment by plispe** [comment]
@bryteise thank you

**Comment by adavidzh** [comment]
> Looks like 19M increase. I enabled the option.

Thanks for this!

**Comment by plispe** [comment]
@bryteise Thank you again for the fix, I installed CL 39980 and It works great
